File: testdata/jennies/builders/map_index_enum/JavaBuilder/sandbox/SomeStructBuilder.java
package sandbox;

import java.util.HashMap;

public class SomeStructBuilder implements cog.Builder<SomeStruct> {
    protected final SomeStruct internal;
    
    public SomeStructBuilder() {
        this.internal = new SomeStruct();
    }
    public SomeStructBuilder data(StringEnum key,String value) {
		if (this.internal.data == null) {
			this.internal.data = new HashMap<>();
		}
        this.internal.data.put(key, value);
        return this;
    }
    public SomeStruct build() {
        return this.internal;
    }
}

File: testdata/jennies/builders/anonymous_struct/CSharpBuilder/src/Grafana/Foundation/AnonymousStruct/SomeStructBuilder.cs
namespace Grafana.Foundation.AnonymousStruct;


public class SomeStructBuilder : Cog.IBuilder<SomeStruct>
{
    protected readonly SomeStruct @internal;

    public SomeStructBuilder()
    {
        this.@internal = new SomeStruct();
    }

    public SomeStructBuilder Time(object time)
    {
        this.@internal.Time = time;
        return this;
    }

    public SomeStruct Build()
    {
        return this.@internal;
    }
}
